app/src/main/java/com/pajato/android/gamechat/chat/fragment/CreateRoomFragment.java

```diff
@@ -89,7 +93,28 @@
     // Protected instance methods.
 
     /** Save the room being created to the Firebase realtime database. */
-    @Override protected void save(@NonNull Account account) {
+    @Override
+    protected boolean save(@NonNull final Account account, final boolean ignoreDupName) {
+        // Determine if the specified name is unique within the current group's rooms
+        if (GroupManager.instance.groupHasRoomName(mGroup.key, mRoom.name) && !ignoreDupName) {
+            dismissKeyboard();
+            new AlertDialog.Builder(getActivity())
```

Comment:
I have seen this alert dialog pattern a half dozen times now.  It strike me that there might be an opportunity to have a helper method take care of this and save a significant number of lines of code.  Just speculation though.